pkg/platform/authentication/auth.go

```diff
@@ -215,6 +217,24 @@ func (s *Auth) AuthenticateWithToken(token string) error {
 	})
 }
 
+// AuthenticateWithDeviceCode authenticates with the given access token obtained via a Platform
+// device authentication request and response.
+func (s *Auth) AuthenticateWithDevice(accessToken *mono_models.JWT) error {
```

Comment:
```suggestion
// AuthenticateWithToken authenticates with the given access token 
func (s *Auth) AuthenticateWithToken(accessToken *mono_models.JWT) error {
```

It doesn't seem like there's any need for this method to have an opinion on where it was obtained.

As part of this could you also update the code where `s.CreateToken()` is currently called? (ie. AuthenticateWithModel). Seems like a sensible change.

Reply:
Okay. I could use some input here though: https://github.com/ActiveState/cli/pull/1716/commits/8fdc40f972c91e70a0f42fa2205718457f62cb4d#diff-e253b887a875b9e415b41ca40c7a25272a7aa49379f529c57ebce4f871fb01d3R185